packages/jupyterlab-manager/src/output.ts

```diff
@@ -107,7 +107,7 @@ class OutputModel extends outputBase.OutputModel {
       break;
     case 'clear_output':
       this.clear_output((msg as KernelMessage.IClearOutputMsg).content.wait);
-      this.set('outputs', [], {newMessage: true});
+      this.set('outputs', this._outputs.toJSON(), {newMessage: true});
       this.save_changes();
```

Comment:
Since the two cases now share the last two lines, you could refactor them down to below the switch statement, i.e., update the internal outputs object, then *always* copy it to the outputs store and save changes.

Reply:
Never mind - that's not a good idea for kernel iopub messages that aren't in the cases.